cli/cli_app.go

```diff
@@ -223,6 +224,46 @@ func runTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 	return runTerragruntWithConfig(terragruntOptions, terragruntConfig, false)
 }
 
+func processBeforeAfterHookActions(beforeAfterHoook []config.BeforeAfterHook, terragruntConfig *config.TerragruntConfig, terragruntOptions *options.TerragruntOptions, previousExecError bool) bool {
+
+	terraformCommand := terragruntOptions.TerraformCliArgs[0]
+
+	errorOccurred := previousExecError
+
+	for i := 0; i < len(beforeAfterHoook); i++ {
+		applicableCommands := beforeAfterHoook[i].Commands
+
+		if contains(applicableCommands, terraformCommand) {
+			//if there's no previous error, execute command
+			//OR if a previos error DID happen AND we want to run anyways
+			//then execute.
+			//Skip execution if there was an error AND we care about errors
+			if !errorOccurred || beforeAfterHoook[i].RunOnError {
+				terragruntOptions.Logger.Printf("Executing hook: %s", beforeAfterHoook[i].Name)
+				actionToExecute := beforeAfterHoook[i].Execute[0]
+				actionParams := beforeAfterHoook[i].Execute[1:]
+				possibleError := shell.RunShellCommand(terragruntOptions, actionToExecute, actionParams...)
+
+				if possibleError != nil {
+					terragruntOptions.Logger.Printf("Error running hook %s with message: %s", beforeAfterHoook[i].Name, possibleError.Error())
+					errorOccurred = true
+				}
+			}
+		}
+	}
+
+	return errorOccurred
+}
+
+func contains(arr []string, str string) bool {
```

Comment:
Stupid Go and its lack of collections. We have this method in `util.ListContainsElement`.